s4e-web/src/app/views/settings/manage-products/manage-products.component.ts

```diff
@@ -0,0 +1,51 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {ProductQuery} from '../../map-view/state/product/product.query';
+import {ProductService} from '../../map-view/state/product/product.service';
+import {Observable} from 'rxjs';
+import {Institution} from '../state/institution/institution.model';
+import {InstitutionsSearchResultsQuery} from '../state/institutions-search/institutions-search-results.query';
+import {ActivatedRoute} from '@angular/router';
+import {untilDestroyed} from 'ngx-take-until-destroy';
+import {ProductLicence} from '../../map-view/state/product/product.model';
+import {map, switchMap, take} from 'rxjs/operators';
+
+@Component({
+  selector: 's4e-settings',
+  templateUrl: './manage-products.component.html',
+  styleUrls: ['./manage-products.component.scss']
+})
+export class ManageProductsComponent implements OnInit, OnDestroy {
+  public areLoading$ = this._productQuery.selectLoading();
+  public errors$ = this._productQuery.selectError();
+  public licences$ = this._productQuery.select('licences');
+
+  public activeInstitution$: Observable<Institution> = this._institutionsSearchResultsQuery
+    .selectActive$(this._activatedRoute);
+
+  constructor(
+    private _productQuery: ProductQuery,
+    private _productService: ProductService,
+    private _institutionsSearchResultsQuery: InstitutionsSearchResultsQuery,
+    private _activatedRoute: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this._productService.getProductsLicences$();
+  }
+
+  toggleInstitutionVisibilityOf(licence: ProductLicence) {
+    this.activeInstitution$
+      .pipe(
+        untilDestroyed(this),
+        take(1),
```

Comment:
If you `take(1)` then this observable completes, `untilDestroyed(this)` is reduntant